### PR TITLE
Audio: Clean up processing function and state of smart amp on component reset

### PR DESCRIPTION
**Pull request by eddy1021**
This PR contributes to issues #4488

### CODE REVIEW

File: src/audio/smart_amp/smart_amp.c [hunk 1]
```diff
@@ -650,8 +650,17 @@ static int smart_amp_copy(struct comp_dev *dev)
...
 {
+	struct smart_amp_data *sad = comp_get_drvdata(dev);
+
 	comp_dbg(dev, "smart_amp_reset()");
 
+	sad->process = NULL;
+	sad->in_channels = 0;
+	sad->out_channels = 0;
+	memset(sad->mod_handle, 0, sizeof(sad->mod_handle));
+	memset(&sad->config, 0, sizeof(sad->config));
```
Comment (cujomalainey) [inline src/audio/smart_amp/smart_amp.c:661]: we don't want to overwrite params from userspace, just variables from params/prepare/copy steps

**Review by cujomalainey — CHANGES_REQUESTED** [review]

File: src/audio/smart_amp/smart_amp.c (hunk 1 above)
Comment (eddy1021) [reply src/audio/smart_amp/smart_amp.c]: Fixed. Thanks!

File: src/audio/smart_amp/smart_amp.c [hunk 2]
```diff
@@ -650,8 +650,14 @@ static int smart_amp_copy(struct comp_dev *dev)
 
 static int smart_amp_reset(struct comp_dev *dev)
 {
+	struct smart_amp_data *sad = comp_get_drvdata(dev);
+
 	comp_dbg(dev, "smart_amp_reset()");
 
+	sad->process = NULL;
```
Comment (lgirdwood) [inline src/audio/smart_amp/smart_amp.c:657]: do we need to free this ?

Comment (eddy1021) [reply src/audio/smart_amp/smart_amp.c:657]: This points to a static function [`smart_amp_process`](https://github.com/thesofproject/sof/blob/main/src/audio/smart_amp/smart_amp.c#L532). It doesn't need to be freed.

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@cujomalainey good for you ?

**Pull request merged**